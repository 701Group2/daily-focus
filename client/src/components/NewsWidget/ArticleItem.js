import React from "react";
import styles from "./style.module.scss";

export default function ArticleItem(props) {
    const article = props;

    if (article) {
        return (
            <div className={styles.articleItem}>
                <div className={styles.articleText}>{article.props.title}</div>

                <div className={styles.articleImage}>
                    <img src={article.props.urlToImage} alt={article.props.description} />
                </div>
            </div>
        );
    } else {
        return "loading..";
    }
}
