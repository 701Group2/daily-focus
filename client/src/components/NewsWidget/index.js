import React, { useState } from "react";
import styles from "./style.module.scss";
import Button from "@material-ui/core/Button";
import { MenuItem, Select } from "@material-ui/core";

function NewsWidget() {
    const [selectedTopic, setSelectedTopic] = useState("sports");
    const [selectedCountry, setsSelectedCountry] = useState("nz");

    return (
        <div className={styles.container}>
            <div className={styles.title}>
                <div className={styles.titleText}>News</div>
            </div>

            <div>News articles go here</div>

            <div className={styles.dropdownsContainer}>
                <Select
                    defaultValue="sports"
                    disableUnderline
                    onChange={(e) => setSelectedTopic(e.target.value)}
                >
                    <MenuItem value="sports">Sports</MenuItem>
                    <MenuItem value="politics">Politics</MenuItem>
                </Select>

                <Select
                    defaultValue="nz"
                    disableUnderline
                    onChange={(e) => setsSelectedCountry(e.target.value)}
                >
                    <MenuItem value="nz">New Zealand</MenuItem>
                    <MenuItem value="usa">USA</MenuItem>
                </Select>
            </div>
        </div>
    );
}

export default NewsWidget;
