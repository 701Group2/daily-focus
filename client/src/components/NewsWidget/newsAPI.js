const NEWS_API_KEY = "91e24cff34754f14aae4b9eaf858855d";

export const getArticles = async (topic, country) => {
    const response = await fetch(
        `https://newsapi.org/v2/top-headlines?category=${topic}&country=${country}&sortBy=publishedAt&apiKey=${NEWS_API_KEY}`
    );
    const json = await response.json();

    return json.articles;
};
