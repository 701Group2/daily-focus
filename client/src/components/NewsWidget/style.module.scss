.container {
    display: flex;
    align-items: center;
    flex-direction: column;
    width: 340px;
    height: fit-content;
    background-color: #ffffff;
    border-radius: 25px;
    box-shadow: 0px 4px 16px 8px rgba(0, 0, 0, 0.1);
}

.dropdownsContainer {
    display: inline-flex;
    margin-bottom: 25px;
}

.title {
    display: flex;
    align-items: center;
    justify-content: center;
    background: #30a0f5;
    height: 48px;
    width: 100%;
    border-top-left-radius: 25px;
    border-top-right-radius: 25px;
}

.titleText {
    color: white;
    font-size: 24px;
    font-family: "Ropa Sans";
}

.countrySelect {
}
