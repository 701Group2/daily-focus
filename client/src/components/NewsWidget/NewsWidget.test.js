import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import { MenuItem, Select } from "@material-ui/core";
import NewsWidget from "./";
import { shallow } from "enzyme";
import { getArticles } from "./newsAPI";

let component;

beforeEach(() => {
    component = shallow(<NewsWidget />);
});

test("it renders correctly", () => {
    const shallowRenderer = new ShallowRenderer();
    const snapshotComponent = shallowRenderer.render(<NewsWidget />);
    expect(snapshotComponent).toMatchSnapshot();
});

test("it contains Material UI Select, MenuItem components", () => {
    expect(component.find(Select)).toHaveLength(2);
    expect(component.find(MenuItem)).toHaveLength(10);
});

test("dropdown default values load in", () => {
    expect(component.find(Select).at(0).props().defaultValue).toEqual("general");
    expect(component.find(Select).at(1).props().defaultValue).toEqual("nz");
});

// checks arbitrary element of array to check it is defined
test("the data is returned from the getArticles function (API endpoint works)", async () => {
    return getArticles("general", "nz").then((data) => {
        expect(data.length).toBeGreaterThan(0);
        expect(typeof data[4].author).toBe("object");
        expect(typeof data[4].content).toBe("string");
        expect(typeof data[4].description).toBe("string");
        expect(typeof data[4].publishedAt).toBe("string");
        expect(typeof data[4].source).toBe("object");
        expect(typeof data[4].source.name).toBe("string");
        expect(typeof data[4].title).toBe("string");
        expect(typeof data[4].url).toBe("string");
        expect(typeof data[4].urlToImage).toBe("string");
    });
});
