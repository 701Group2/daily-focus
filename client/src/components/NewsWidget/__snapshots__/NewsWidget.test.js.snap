// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="title"
  >
    <div
      className="titleText"
    >
      News
    </div>
  </div>
  <div
    className="articleList"
  >
    <div
      className="loading"
    >
       Loading... 
    </div>
  </div>
  <div
    className="dropdownsContainer"
  >
    <WithStyles(ForwardRef(Select))
      className="dropdown"
      defaultValue="general"
      disableUnderline={true}
      onChange={[Function]}
    >
      <WithStyles(ForwardRef(MenuItem))
        value="sports"
      >
        Sports
      </WithStyles(ForwardRef(MenuItem))>
      <WithStyles(ForwardRef(MenuItem))
        value="business"
      >
        Business
      </WithStyles(ForwardRef(MenuItem))>
      <WithStyles(ForwardRef(MenuItem))
        value="health"
      >
        Health
      </WithStyles(ForwardRef(MenuItem))>
      <WithStyles(ForwardRef(MenuItem))
        value="entertainment"
      >
        Entertainment
      </WithStyles(ForwardRef(MenuItem))>
      <WithStyles(ForwardRef(MenuItem))
        value="science"
      >
        Science
      </WithStyles(ForwardRef(MenuItem))>
      <WithStyles(ForwardRef(MenuItem))
        value="technology"
      >
        Tech
      </WithStyles(ForwardRef(MenuItem))>
      <WithStyles(ForwardRef(MenuItem))
        value="general"
      >
        General
      </WithStyles(ForwardRef(MenuItem))>
    </WithStyles(ForwardRef(Select))>
    <WithStyles(ForwardRef(Select))
      className="dropdown"
      defaultValue="nz"
      disableUnderline={true}
      onChange={[Function]}
    >
      <WithStyles(ForwardRef(MenuItem))
        value="nz"
      >
        New Zealand
      </WithStyles(ForwardRef(MenuItem))>
      <WithStyles(ForwardRef(MenuItem))
        value="us"
      >
        USA
      </WithStyles(ForwardRef(MenuItem))>
      <WithStyles(ForwardRef(MenuItem))
        value="au"
      >
        Australia
      </WithStyles(ForwardRef(MenuItem))>
    </WithStyles(ForwardRef(Select))>
  </div>
</div>
`;
